### PR TITLE
Memory Optimization for Compilation in OVEP 

### PR DESCRIPTION
**Pull request by sfatimar**
Calling Split API Calls Read+Model in lieu of unified Compile Model call for export compile flow to ensure memory optimization. Freeing up model proto and serialized string and read model ov ir later to free up memory for the ahead pipeline
Optimization during EpCtxt flow
All the Graph related operations require all the Node Attributes to be set while dealing with model instances internally with them, in the existing implementation these attributes make a copy when constructing a Graph dynamically during runtime.
Propose to use these attributes in place without creating a copy to avoid memory allocation / copy while calling these Graph related functions.
Changes to ensure the bug fixes related to openvino version and epctxt file path.
Moving Compiler version to C++20 for getting r-value mem optimizations benefit

### Motivation and Context
This change is required because memory optimization during Compilation flow is too high.

### CODE REVIEW

**Comment by sfatimar** [comment]
@jywu-msft

**Comment by sfatimar** [comment]
@jywu-msft Please review

**Comment by sfatimar** [comment]
Fixed Unit test failures

File: onnxruntime/core/providers/openvino/backend_manager.cc [hunk 1]
```diff
@@ -155,32 +159,28 @@ Status BackendManager::ExportCompiledBlobAsEPCtxNode(const onnxruntime::GraphVie
...
   }
+
   // If embed_mode, then pass on the serialized blob
   // If not embed_mode, dump the blob here and only pass on the path to the blob
   if (global_context_.ep_context_embed_mode) {
     std::ostringstream model_blob_stream;
     compiled_model.export_model(model_blob_stream);
-    model_blob_str = model_blob_stream.str();
-    ORT_ENFORCE(model_blob_str.size() != 0);
+    model_blob_str = std::move(model_blob_stream).str();
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/openvino/backend_manager.cc:177]: Is this the main reason for C++ 20 (to move the string from ostringstream)?

File: onnxruntime/core/providers/openvino/backends/basic_backend.cc [hunk 2]
```diff
@@ -91,26 +89,36 @@ BasicBackend::BasicBackend(const ONNX_NAMESPACE::ModelProto& model_proto,
                                                            device_config,
                                                            global_context_.ep_context_embed_mode,
                                                            subgraph_context_.subgraph_name);
-        ie_cnn_network_ = exe_network_.Get().get_runtime_model();
+        // ie_cnn_network_ = exe_network_.Get().get_runtime_model();
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/openvino/backends/basic_backend.cc:92]: nit: Just want to make sure that commenting-out this line won't cause any issues.

**Review by adrianlizarraga — APPROVED** [review]
LGTM. Just had a minor question.

File: onnxruntime/core/providers/openvino/backend_manager.cc (hunk 1 above)
Comment (javier-intel) [reply onnxruntime/core/providers/openvino/backend_manager.cc:177]: That's correct, c++20 introduces an str() overload that returns an r-value reference. More details below
https://en.cppreference.com/w/cpp/io/basic_ostringstream/str
https://stackoverflow.com/questions/26266525/move-the-string-out-of-a-stdostringstream

File: onnxruntime/core/providers/openvino/backends/basic_backend.cc (hunk 2 above)
Comment (javier-intel) [reply onnxruntime/core/providers/openvino/backends/basic_backend.cc:92]: Hmm, this looks like a merge issue. The PR will be updated to revert this change. Thanks for the eagle eyes.

**Comment by preetha-intel** [comment]
@jywu-msft  Can you help with retriggering the linux pipelines for the failure logs on onnxruntime_test_python.py unit tests.

File: onnxruntime/test/python/onnxruntime_test_python.py [hunk 3]
```diff
@@ -1691,6 +1691,8 @@ def test_register_custom_e_ps_library(self):
         # skip amd gpu build
         if "kRocmExecutionProvider" in available_eps:
             return
+        if "kOpenVINOExecutionProvider" in available_eps:
```
Comment (jywu-msft) [inline onnxruntime/test/python/onnxruntime_test_python.py:1694]: lines 1692 and 1694 have a problem. it should be 'ROCMExecutionProvider' and 'OpenVINOExecutionProvider'

**Comment by preetha-intel** [comment]
@jywu-msft  The merge conflicts are fixed. OVEP Internal ci pipeline is successful. Can you kick start the workflows at your availability

**Comment by jywu-msft** [comment]
> @jywu-msft The merge conflicts are fixed. OVEP Internal ci pipeline is successful. Can you kick start the workflows at your availability

seems like it is still crashing on a unit test. wonder if it's another test after the one that's skipped ? maybe we need to enable verbose logging again unless you an repro from your end.

**Comment by jywu-msft** [comment]
please also fix the clang-format lint error https://github.com/microsoft/onnxruntime/actions/runs/10683693892/job/29618758615?pr=21872

**Comment by preetha-intel** [comment]
> please also fix the clang-format lint error https://github.com/microsoft/onnxruntime/actions/runs/10683693892/job/29618758615?pr=21872

Fixed.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**